keps/sig-node/4800-cpumanager-split-l3cache/README.md

```diff
@@ -0,0 +1,1235 @@
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+We proposed this KEP to introduce a new CPU Manager static policy option, "align-cpus-by-uncorecache", that groups CPU resources by last level cache where possible.
+The opt-in feature changes the cpu assignment algorithm to add sorting by L3 cache and then taking cpus aligned to the same L3 cache, where possible.
+In cases where numbers of cpu requested exceeds number of cpus grouped in the same L3 cache, the algorithm attempts best-effort to reduce assignments of cpus across minimal numbers of L3s.
+If the cpumanager can’t align optimally, it will still admit the workload as before.  More details will be given in the motivation section.
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+Our motivation is to reduce noisy neighbor scenarios that occur on systems with split L3 cache, which is available on both x86 and ARM architecture.
+The challenge with current kubelet’s cpu manager is that it is unaware of split L3 architecture and spreads CPU assignments across distributed L3s. This creates a noisy neighbor problem where multiple pods/containers are sharing the same L3 cache. In addition, pods spread against multiple L3 cache sees higher latency and reduced performance due to inter-cache access latency. For workload use cases that are sensitive to latency and are performance deterministic, minimizing the noisy neighbor condition in L3 can have significant improvements in performance.
+
+![Figure to the motivation](https://github.com/ajcaldelas/enhancements/blob/c7e423439608001cc837e2f5629dc87935353611/keps/sig-node/4800-cpumanager-split-l3cache/images/l3-alignment-motivation.png)
+
+The figure below highlights performance gain when pod placement is aligned to an L3 group. In this exampleTPCC/My-SQL  is the workload deployed. There was a 18% uplift compared to default behavior.  Other workloads might see  higher gains.  For Stream workload, highlighted in the original design doc, the performance uplift was approximately 20%.
```

Comment:
```suggestion
The figure below highlights performance gain when pod placement is aligned to an L3 group. In this exampleTPCC/My-SQL  is the workload deployed. There was a 18% uplift compared to default behavior.  Other workloads might see  higher gains.
```

I would either mention the stream use case here or drop the phrase. Who knows if the design doc will be relevant so I would view this KEP as the design doc from now on.

Reply:
I agree. KEPs almost always supercede and replace design docs.

Reply:
Ill remove this and mention this to the team to see how we want to approach this